lw/net/tls.cpp

```diff
@@ -0,0 +1,109 @@
+#include "lw/net/tls.h"
+
+#include <memory>
+
+#include "lw/flags/flags.h"
+#include "lw/memory/buffer.h"
+#include "lw/memory/buffer_view.h"
+#include "lw/net/internal/tls_client.h"
+#include "lw/net/internal/tls_context.h"
+
+LW_FLAG(
+  std::size_t, tls_buffer_size, 1024 * 1024,
+  "Size of write buffer for TLS streams."
+);
+
+namespace lw::net {
+
+TLSStream::TLSStream(
+  std::unique_ptr<internal::TLSClientImpl> client,
+  std::unique_ptr<io::CoStream> raw_stream
+):
+  _client{std::move(client)},
+  _raw_stream{std::move(raw_stream)},
+  _write_buffer{flags::tls_buffer_size}
+{}
+
+TLSStream::~TLSStream() {
+  // This destructor must be defined in this file where it has access to the
+  // full definition of the TLS*Impl classes.
+}
+
+co::Future<void> TLSStream::handshake() {
+  // Get into handshake mode.
+  _client->handshake();
```

Comment:
Add documentation to `TLSClientImple::handshake` about it being safe (and intended) that you call it repeatedly.